lib/Sema/TypeCheckRequestFunctions.cpp

```diff
@@ -25,13 +25,13 @@ Type InheritedTypeRequest::evaluate(
                         llvm::PointerUnion<TypeDecl *, ExtensionDecl *> decl,
                         unsigned index) const {
   // Figure out how to resolve types.
-  TypeResolutionOptions options;
+  TypeResolutionOptions options = None;
   DeclContext *dc;
   if (auto typeDecl = decl.dyn_cast<TypeDecl *>()) {
     if (auto nominal = dyn_cast<NominalTypeDecl>(typeDecl)) {
       dc = nominal;
-      options |= TypeResolutionFlags::GenericSignature;
-      options |= TypeResolutionFlags::InheritanceClause;
+      options = TypeResolutionOptions(TypeResolverContext::GenericSignature);
+      options.setContext(TypeResolverContext::InheritanceClause);
```

Comment:
Once again you're setting the context twice?

Reply:
See the earlier explanation.